### ISSUE TITLE
Proguard issue

### ISSUE THREAD

**Issue by kisipila** [urbanairship/android-library#5]
Hi,
I'm wondering the following Proguard rule:

`-keepnames class * implements android.os.Parcelable {
    public static final ** CREATOR;
}`

It, as I have understood it, prevents the obfuscation of any parcelable class - not only in the library but also in the application using Urbanairship. I would expect to see using keepclassmembers/keepclassmembernames instead or limit the scope of the rule to Urbanairship units only.

**Comment by rlepinski** [comment]
The default android proguard contains:
```
-keepclassmembers class * implements android.os.Parcelable {
  public static final android.os.Parcelable$Creator CREATOR;
}
```

It looks slightly different than what we currently have, but I am not sure if the end result is any different.

**Comment by kisipila** [comment]
Yes, the end result is different. The current implementation prevents the obfuscation of both the class name and the class member field. The default implementation allows the obfuscation of the class name but keeps the class member intact. As I said above the impact is not limited to the library alone but affects the whole application.

**Comment by rlepinski** [comment]
@kisipila we will patch it next week in 8.2.3

**Comment by kisipila** [comment]
Excellent, thank you.

**Comment by rlepinski** [comment]
Released 8.2.3

Thanks for the report!
